diff --git a/src/tau2/data_model/tasks.py b/src/tau2/data_model/tasks.py
index 28f3788..b91b97e 100644
--- a/src/tau2/data_model/tasks.py
+++ b/src/tau2/data_model/tasks.py
@@ -12,7 +12,7 @@ from typing_extensions import Annotated
 from tau2.data_model.message import Message, ToolCall, ToolRequestor
 
 
-class StructuredUserInstructions(BaseModel):
+class StructuredUserInstructions(BaseModel):#BaseModel是pydantic库中的一个类，用于创建数据模型。它提供了数据验证、序列化和反序列化等功能。通过继承BaseModel，StructuredUserInstructions类可以定义一组字段，并自动生成验证逻辑和其他有用的方法。
     """
     User instructions. This information defines the specific situation the user is in and the tasks they are trying to complete.
     """
@@ -263,7 +263,7 @@ class RewardType(str, Enum):
     DB = "DB"
     ENV_ASSERTION = "ENV_ASSERTION"
     NL_ASSERTION = "NL_ASSERTION"
-    ACTION = "ACTION"
+    ACTION = "ACTION"#有action_bais的任务，要求agent必须按照actions中的action顺序执行，否则无法获得奖励
     COMMUNICATE = "COMMUNICATE"
 
 
@@ -654,7 +654,7 @@ def make_task(
     initialization_data: Optional[InitializationData] = None,
     initialization_actions: Optional[list[EnvFunctionCall]] = None,
     message_history: Optional[list[Message]] = None,
-) -> Task:
+) -> Task:#基于用户指令、评估标准和消息历史创建一个任务对象
     """
     Make a task from a user instruction, an evaluation criteria and a message history.
     """
diff --git a/src/tau2/domains/airline/environment.py b/src/tau2/domains/airline/environment.py
index e1cabff..6ac2f33 100644
--- a/src/tau2/domains/airline/environment.py
+++ b/src/tau2/domains/airline/environment.py
@@ -14,7 +14,7 @@ from tau2.environment.environment import Environment
 from tau2.utils import load_file
 
 
-def get_environment(
+def get_environment( ##加载数据库，工具和策略 
     db: Optional[FlightDB] = None,
     solo_mode: bool = False,
 ) -> Environment:
@@ -34,7 +34,7 @@ def get_environment(
 
 def get_tasks(task_split_name: Optional[str] = "base") -> list[Task]:
     tasks = load_file(AIRLINE_TASK_SET_PATH)
-    tasks = [Task.model_validate(task) for task in tasks]
+    tasks = [Task.model_validate(task) for task in tasks]#把task字典转换为Task对象
     if task_split_name is None:
         return tasks
     task_splits = get_tasks_split()
diff --git a/src/tau2/environment/environment.py b/src/tau2/environment/environment.py
index d9dd3be..a638911 100644
--- a/src/tau2/environment/environment.py
+++ b/src/tau2/environment/environment.py
@@ -60,7 +60,7 @@ class Environment:
         self.solo_mode = solo_mode
         if self.solo_mode:
             self.validate_solo_mode()
-        self.sync_tools()
+        self.sync_tools()#默认什么都不做，但有些domin会重写用来同步agentDB和userDB
 
     def get_domain_name(self) -> str:
         """
diff --git a/src/tau2/environment/toolkit.py b/src/tau2/environment/toolkit.py
index b6ccc92..45571c8 100644
--- a/src/tau2/environment/toolkit.py
+++ b/src/tau2/environment/toolkit.py
@@ -9,7 +9,7 @@ from tau2.utils import get_dict_hash, update_pydantic_model_with_dict
 
 TOOL_ATTR = "__tool__"
 TOOL_TYPE_ATTR = "__tool_type__"
-MUTATES_STATE_ATTR = "__mutates_state__"
+MUTATES_STATE_ATTR = "__mutates_state__"#是否会改变环境状态的属性，True表示会改变，False表示不会改变
 DISCOVERABLE_ATTR = "__discoverable__"
 
 
diff --git a/src/tau2/gym/gym_agent.py b/src/tau2/gym/gym_agent.py
index e7cd1e2..d09e993 100644
--- a/src/tau2/gym/gym_agent.py
+++ b/src/tau2/gym/gym_agent.py
@@ -614,11 +614,11 @@ class AgentGymEnv(gym.Env):
         self._lock = threading.Lock()
         self._agent: Optional[GymAgent] = None
         self._user: Optional[UserSimulator] = None
-        self._orchestrator: Optional[Orchestrator] = None
+        self._orchestrator: Optional[Orchestrator] = None#？
         self._orchestrator_thread: Optional[threading.Thread] = None
-        self._simulation_done = threading.Event()
-        self._simulation_run: Optional[SimulationRun] = None
-        self.observation_space = TauSpace()
+        self._simulation_done = threading.Event()#
+        self._simulation_run: Optional[SimulationRun] = None#保存整个模拟的结果
+        self.observation_space = TauSpace()#本质上是一个字符串空间，表示环境的观察值是一个字符串
         self.action_space = TauSpace()
 
     def _get_tools(self) -> List[Tool]:
